### PR TITLE
recipe for karllark/dust_extinction

### PR DESCRIPTION
**Pull request by karllark**
Hopefully straightforward PR for a pure python package.

### CODE REVIEW

File: recipes/dust_extinction/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: c038d51d94dc65a4c6e3f473e60888f79e65e5acb07af49598012823499f34fa
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  build:
+    - astropy >=1.3
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]: All of these `build` requirements should be in `host` section.

Comment (karllark) [reply recipes/dust_extinction/meta.yaml]: Using a different template now.  So, maybe fixed?

File: recipes/dust_extinction/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  fn: '{{ name }}-{{ version }}.tar.gz'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 1000
+  script: python setup.py install --offline --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]:
```suggestion
  host:
```

File: recipes/dust_extinction/meta.yaml (hunk 1 above)
Comment (chrisburr) [reply recipes/dust_extinction/meta.yaml]: I've left another comment with the change that should make the linter happy.

File: recipes/dust_extinction/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    - python
+    - setuptools
+    - astropy >=1.3
```
Comment (pllim) [inline recipes/dust_extinction/meta.yaml]:
```suggestion
    - astropy >=2
```

File: recipes/dust_extinction/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  run:
+    - python
+    - numpy >=1.6
```
Comment (pllim) [inline recipes/dust_extinction/meta.yaml]:
```suggestion
    - numpy >=1.13
```

File: recipes/dust_extinction/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    - scipy
+    - astropy >=1.3
```
Comment (pllim) [inline recipes/dust_extinction/meta.yaml]:
```suggestion
    - astropy >=2
```

**Comment by karllark** [comment]
I do not understand why the win_64 failed.  It is not clear from the logs.  Any advice from anyone?

**Comment by pllim** [comment]
Took a few clicks to get to the raw log:
```
2019-02-27T14:22:55.9100753Z ##[section]Starting: Build recipe (maybe vs2008)
2019-02-27T14:22:55.9196265Z ==============================================================================
2019-02-27T14:22:55.9196358Z Task         : Command Line
2019-02-27T14:22:55.9196407Z Description  : Run a command line script using cmd.exe on Windows and bash on macOS and Linux.
2019-02-27T14:22:55.9196454Z Version      : 2.146.1
2019-02-27T14:22:55.9196529Z Author       : Microsoft Corporation
2019-02-27T14:22:55.9196576Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613735)
2019-02-27T14:22:55.9196625Z ==============================================================================
2019-02-27T14:22:56.7059943Z Generating script.
2019-02-27T14:22:56.7756985Z ##[command]"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\77bf86bd-b4e8-4ca4-8898-c6de1de1f2ac.cmd""
2019-02-27T14:23:19.3431901Z Setting build arch. This is only useful when pretending to be on another arch, such as for rendering necessary dependencies on a non-native arch. I trust that you know what you're doing.
2019-02-27T14:23:19.3460157Z Setting build platform. This is only useful when pretending to be on another platform, such as for rendering necessary dependencies on a non-native platform. I trust that you know what you're doing.
2019-02-27T14:23:22.1532049Z No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
2019-02-27T14:23:22.1533454Z WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
2019-02-27T14:23:22.2365329Z Building dust_extinction with conda-forge/label/main
2019-02-27T14:23:22.2365703Z INFO:conda_build.variants:Adding in variants from internal_defaults
2019-02-27T14:23:22.2365777Z rendering D:\a\1\s\recipes\dust_extinction for win-win-64
2019-02-27T14:23:22.2365971Z Computed that there are 3 distributions to build from 1 recipes
2019-02-27T14:23:22.2366162Z Resolved dependencies, will be built in the following order:
2019-02-27T14:23:22.2366349Z     dust_extinction-0.7-python_2.7-target_win-64-on-win-win-64
2019-02-27T14:23:22.2366531Z     dust_extinction-0.7-python_3.7-target_win-64-on-win-win-64
2019-02-27T14:23:22.2366709Z     dust_extinction-0.7-python_3.6-target_win-64-on-win-win-64
2019-02-27T14:23:22.2367126Z Adding in variants from internal_defaults
2019-02-27T14:23:22.2367415Z Adding in variants from C:\Miniconda\conda_build_config.yaml
2019-02-27T14:23:22.2368010Z INFO:conda_build.variants:Adding in variants from C:\Miniconda\conda_build_config.yaml
2019-02-27T14:23:22.5656569Z INFO:conda_build.metadata:Attempting to finalize metadata for dust_extinction
2019-02-27T14:23:22.5657033Z Attempting to finalize metadata for dust_extinction
2019-02-27T14:23:22.7894373Z Traceback (most recent call last):
2019-02-27T14:23:22.7898170Z   File ".ci_support\build_all.py", line 130, in <module>
2019-02-27T14:23:22.7898262Z     build_all(args.recipes_dir, args.arch)
2019-02-27T14:23:22.7898354Z   File ".ci_support\build_all.py", line 68, in build_all
2019-02-27T14:23:22.7898414Z     build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
2019-02-27T14:23:22.7898494Z   File ".ci_support\build_all.py", line 120, in build_folders
2019-02-27T14:23:22.7898556Z     conda_build.api.build(list(d.keys()), config=get_config(arch, channel_urls))
2019-02-27T14:23:22.7898632Z   File "C:\Miniconda\lib\site-packages\conda_build\api.py", line 208, in build
2019-02-27T14:23:22.7898715Z     notest=notest, need_source_download=need_source_download, variants=variants)
2019-02-27T14:23:22.7898779Z   File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 2314, in build_tree
2019-02-27T14:23:22.7898834Z     notest=notest,
2019-02-27T14:23:22.7898894Z   File "C:\Miniconda\lib\site-packages\conda_build\build.py", line 1324, in build
2019-02-27T14:23:22.7898980Z     output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
2019-02-27T14:23:22.7899044Z   File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 735, in expand_outputs
2019-02-27T14:23:22.7900130Z     for (output_dict, m) in _m.copy().get_output_metadata_set(permit_unsatisfiable_variants=False):
2019-02-27T14:23:22.7900201Z   File "C:\Miniconda\lib\site-packages\conda_build\metadata.py", line 2038, in get_output_metadata_set
2019-02-27T14:23:22.7900265Z     bypass_env_check=bypass_env_check)
2019-02-27T14:23:22.7900367Z   File "C:\Miniconda\lib\site-packages\conda_build\metadata.py", line 721, in finalize_outputs_pass
2019-02-27T14:23:22.7900429Z     permit_unsatisfiable_variants=permit_unsatisfiable_variants)
2019-02-27T14:23:22.7900492Z   File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 509, in finalize_metadata
2019-02-27T14:23:22.7900566Z     exclude_pattern)
2019-02-27T14:23:22.7900627Z   File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 371, in add_upstream_pins
2019-02-27T14:23:22.7900685Z     permit_unsatisfiable_variants, exclude_pattern)
2019-02-27T14:23:22.7900773Z   File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 359, in _read_upstream_pin_files
2019-02-27T14:23:22.7900835Z     permit_unsatisfiable_variants=permit_unsatisfiable_variants)
2019-02-27T14:23:22.7900898Z   File "C:\Miniconda\lib\site-packages\conda_build\render.py", line 143, in get_env_dependencies
2019-02-27T14:23:22.7900976Z     channel_urls=tuple(m.config.channel_urls))
2019-02-27T14:23:22.7901180Z   File "C:\Miniconda\lib\site-packages\conda_build\environ.py", line 738, in get_install_actions
2019-02-27T14:23:22.7901240Z     locking=locking, timeout=timeout)
2019-02-27T14:23:22.7901321Z   File "C:\Miniconda\lib\site-packages\conda_build\index.py", line 241, in get_build_index
2019-02-27T14:23:22.7901377Z     platform=subdir)
2019-02-27T14:23:22.7901435Z   File "C:\Miniconda\lib\site-packages\conda\exports.py", line 243, in get_index
2019-02-27T14:23:22.7901519Z     index = _get_index(channel_urls, prepend, platform, use_local, use_cache, unknown, prefix)
2019-02-27T14:23:22.7901592Z   File "C:\Miniconda\lib\site-packages\conda\common\io.py", line 85, in decorated
2019-02-27T14:23:22.7901647Z     return f(*args, **kwds)
2019-02-27T14:23:22.7901724Z   File "C:\Miniconda\lib\site-packages\conda\core\index.py", line 58, in get_index
2019-02-27T14:23:22.7901784Z     index = fetch_index(channel_urls, use_cache=use_cache)
2019-02-27T14:23:22.7901852Z   File "C:\Miniconda\lib\site-packages\conda\core\index.py", line 73, in fetch_index
2019-02-27T14:23:22.7901927Z     sd = SubdirData(Channel(url))
2019-02-27T14:23:22.7901989Z   File "C:\Miniconda\lib\site-packages\conda\core\subdir_data.py", line 60, in __call__
2019-02-27T14:23:22.7902045Z     assert channel.subdir
2019-02-27T14:23:22.7902094Z AssertionError
2019-02-27T14:23:23.7023807Z ##[error]Cmd.exe exited with code '1'.
2019-02-27T14:23:23.7414490Z ##[section]Finishing: Build recipe (maybe vs2008)
```

**Comment by pllim** [comment]
Is this related? The default numpy 1.11 seems a bit too old.
```
WARNING:conda_build.metadata:No numpy version specified in conda_build_config.yaml.  Falling back to default numpy value of 1.11
```

**Comment by chrisburr** [comment]
You can ignore `win_64` on Azure, it's known to be broken (#7715). Appveyor is currently provided Windows tests.

File: recipes/dust_extinction/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.tar.gz'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 1000
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]: The build number should start from `0` for new versions.
```suggestion
  number: 0
```

File: recipes/dust_extinction/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+{% set version = "0.7" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "68c096e60b133cfaa60e641cd6a1c9b45a0f9b82d2cd4c8560ce68db3182373d" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.tar.gz'
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]: This line isn't needed.
```suggestion
```

**Comment by karllark** [comment]
@chrisburr : great.  

Not sure what is up with the osx_64 osx build.  The details say there was a problem and the build has been abandoned.  So, not sure what is the next step here.

File: recipes/dust_extinction/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 1000
+  script: python setup.py install --offline --single-version-externally-managed --record=record.txt
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]:
```suggestion
  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```

File: recipes/dust_extinction/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  fn: '{{ name }}-{{ version }}.tar.gz'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]: Could this be a [`noarch: python`](http://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-builds) package?
```suggestion
build:
  noarch: python
```

File: recipes/dust_extinction/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+    - matplotlib
+
+test:
+  imports:
+    - dust_extinction
+
+about:
+  home: https://github.com/karllark/dust_extinction
+  license: BSD 3-clause
+  license_family: BSD
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]:
```suggestion
  license_family: BSD
  license_file: licenses/LICENSE.rst
```

File: recipes/dust_extinction/meta.yaml (hunk 9 above)
Comment (karllark) [reply recipes/dust_extinction/meta.yaml]: yes!

File: recipes/dust_extinction/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
+
+requirements:
+  host:
+    - python
```
Comment (chrisburr) [inline recipes/dust_extinction/meta.yaml]: You need `pip` here:
```suggestion
    - python
    - pip
```

Comment (karllark) [reply recipes/dust_extinction/meta.yaml]: Thanks.

**Comment by karllark** [comment]
Why is everything failing now when most were working before?  This seems bad.

**Comment by chrisburr** [comment]
@karllark I think [it's caused by `astropy_helpers` trying to auto-update](https://github.com/astropy/conda-channel-astropy/blob/master/requirements.yml#L22-L26).

I've pushed a patch which disables this behaviour so you can still use `pip` to install. If you would rather fix it a different way feel free to revert my commit.

**Comment by pllim** [comment]
Hmm @mwcraig ?

**Comment by mwcraig** [comment]
I can take a look Sunday — the helpers auto update is really a bit of a hassle.

**Comment by mwcraig** [comment]
@chrisburr's patch looks fine; the alternative is to patch [`setup.cfg` here](https://github.com/karllark/dust_extinction/blob/master/setup.cfg#L22) to `auto_use = False`. I'd say merge this as-is and then, if desired, change the recipe in the feedstock. 

I restarted the windows build that failed, hoping it was something transient.

**Comment by mwcraig** [comment]
@conda-forge/staged-recipes -- any idea what the source fo this failure might be? https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=18495&view=logs&jobId=3ff94dba-189a-527c-65e3-ce8503824159&taskId=3732236d-eeaf-5163-566e-5f7a7d44f993&lineStart=68&lineEnd=69&colStart=1&colEnd=1

**Comment by chrisburr** [comment]
@mwcraig Thanks for taking a look.

The Azure `win_64` build is known to be broken for staged-recipes (#7715) but Appveyor provides testing for Windows.

**Pull request merged**